src/arrayMethodJoin.js

```diff
@@ -4,9 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const length = this.length;
+    let resultJoin = '';
+    let counter = 0;
+
+    while (counter < length) {
+      if (counter > 0) {
+        resultJoin += `${separator}`;
+      }
+
+      let next = '';
+      const elem = this[counter];
+
+      next = elem === null || elem === undefined ? '' : elem;
+      resultJoin += `${next}`;
```

Comment:
no interpolation here, also
```suggestion
      resultJoin += next;
```